platform/os.go

```diff
@@ -0,0 +1,70 @@
+package platform
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"log"
+	"os"
+)
+
+// ReadFileByLines reads file line by line and return array of lines.
+func ReadFileByLines(filename string) ([]string, error) {
+	var (
+		lineStrArr []string
+	)
+
+	f, err := os.Open(filename)
+	if err != nil {
+		return nil, fmt.Errorf("Error opening %s file error %v", filename, err)
+	}
+
+	defer f.Close()
+
+	r := bufio.NewReader(f)
+
+	for {
+		lineStr, err := r.ReadString('\n')
+		if err != nil {
+			if err != io.EOF {
+				return nil, fmt.Errorf("Error reading %s file error %v", filename, err)
+			}
+
+			lineStrArr = append(lineStrArr, lineStr)
+			break
+		}
+
+		lineStrArr = append(lineStrArr, lineStr)
+	}
+
+	return lineStrArr, nil
+}
+
+func CheckIfFileExists(filepath string) (bool, error) {
+	_, err := os.Stat(filepath)
+	if err == nil {
+		return true, nil
+	}
+
+	if os.IsNotExist(err) {
+		return false, nil
+	}
+
+	return true, err
+}
+
+func CreateDirectory(dirPath string) error {
+	var err error
+
+	if dirPath == "" {
+		log.Printf("dirPath is empty, nothing to create.")
+		return nil
+	}
+
+	isExist, _ := CheckIfFileExists(dirPath)
```

Comment:
check error?

Reply:
same here...i can print err but i cannot validate error

Reply:
checked this link: https://stackoverflow.com/questions/37932551/mkdir-if-not-exists-using-golang . let me validate error also and createdirectory